caffe2/operators/hip/pool_op_miopen.hip

```diff
@@ -34,16 +34,12 @@ class MIOPENPoolOp : public ConvPoolOpBase<HIPContext> {
     MIOPEN_ENFORCE(miopenCreateTensorDescriptor(&top_desc_));
     MIOPEN_ENFORCE(miopenCreatePoolingDescriptor(&pooling_desc_));
 
-    if ((operator_def.type().substr(0, 9) == "MaxPool") ||
-        (operator_def.type().substr(0, 17) == "MaxPoolGradient") ||
-        (operator_def.type().substr(0, 9) == "MaxPool3D") ||
-        (operator_def.type().substr(0, 17) == "MaxPool3DGradient"))  {
+    if ((operator_def.type().substr(0, 7) == "MaxPool") ||
+        (operator_def.type().substr(0, 15) == "MaxPoolGradient")) {
```

Comment:
The `MaxPoolGradient` is redundant, but not harmful.   The substr(0,7) == "MaxPool" will also cover the "MaxPoolGradient" case since "MaxPoolGradient" starts with "MaxPool".  Right?

Reply:
Yeah that makes sense, let me change the old code and retest it.